### DISCUSSION TITLE
How do I compile on Windows

### DISCUSSION THREAD

**Discussion by xiuer-1573** [alvr-org/ALVR#1010] [Q&A]
i have download The source code, How do I compile on Windows?

**Comment by thisleo** [comment]
follow [this](https://github.com/alvr-org/ALVR/wiki/Building-From-Source)

**Reply by xiuer-1573** [reply]
thank you
I've installed vscode and rust-analyzer, and I just want to compile the server separately. What do I do

**Reply by thisleo** [reply]
you need install 
1. rustup 
2. visual studio
3. LLVM
4. visualstudio2019buildtools  (The version must be the same as visual studio)

**Comment by xiuer-1573** [comment]


**Reply by xiuer-1573** [reply]
thank you
I've installed vscode and rust-analyzer, and I just want to compile the server separately. What do I do

**Reply by zmerp** [reply]
The command is just `cargo xtask build-server`. Add `--release` to build as release. If you build the server on the master branch it will be compatible with the nightly client, if you build on the v18 branch it will be compatible with the stable client.

**Reply by xiuer-1573** [reply]
i just install vscode and rust-analyzer,can i do that? Or do I need to use visual studio 2019 and LLVM

**Reply by zmerp** [reply]
You do not need Visual Studio, only the build tools with MSVC. You need to also install LLVM.

**Reply by zmerp** [reply]
You are better off following the guide step by step: https://github.com/alvr-org/ALVR/wiki/Building-From-Source#windows
This is for building the master branch.

**Reply by xiuer-1573** [reply]
thank you very much,i have build successful
